plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/jdbc/cache/JDBCCompositeCache.java

```diff
@@ -390,27 +390,25 @@ protected void loadObjects(DBRProgressMonitor monitor, OWNER owner, PARENT forPa
         // Fill global cache
         synchronized (this) {
             synchronized (objectCache) {
-                if (forParent != null || !parentObjectMap.isEmpty()) {
-                    if (forParent == null) {
-                        // Cache global object list
-                        List<OBJECT> globalCache = new ArrayList<>();
-                        for (Map<String, ObjectInfo> objMap : parentObjectMap.values()) {
-                            if (objMap != null) {
-                                for (ObjectInfo info : objMap.values()) {
-                                    if (!info.broken) {
-                                        globalCache.add(info.object);
-                                    }
+                if (forParent == null) {

```

Comment:
I'm not sure it's ok to just throw away the condition

Reply:
Well, the `forParent != null` never actually did anything because, if you look at how the logic was structured, we checked `forParent != null || !parentObjectMap.isEmpty()`, then right after that, we checked `forParent == null` which makes the first part of the condition invalid. 

`parentObjectMap` is filled only if the value didn't exist in its true parent, but we don't actually need it if we just want to update our global cache from 
```
                    for (List<OBJECT> objects : objectCache.values()) {
                        globalCache.addAll(objects);
                    }
```

Reply:
Thank you for explanation, it became clearer. I hope you're right